core/utility.ml

```diff
@@ -982,6 +982,8 @@ let getenv : string -> string option =
 let safe_getenv s =
   try Sys.getenv s
   with NotFound _ ->
+    (* We need to retain this `failwith` since `errors.ml` depends on
```

Comment:
Does OCaml have a way to break module cycles?

Reply:
Yes. You can always break a cycle by introducing another layer of indirection, for example by functoring the module.

Morally, I don't think we should use the `errors` module here anyways, as errors originating from the `utility` module ought to be interpreted at the use sites, rather than directly by the definition of `utility`. What we really should do, is to define and export meaningful exceptions in the `utility` module that may be raised by utility functions.